diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -43,76 +51,127 @@ class BlockStructureBraces : Rule("block-structure") {
+        }
+        if (hasElseBranch && elseNode!!.elementType != IF && elseNode.hasChildOfType(LBRACE)) {
+            checkOpenBrace(node, ELSE, configuration)
+            checkCloseBrace(elseNode, configuration)
         }
     }
 
-    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType) {
+    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
+        if (!configuration.openBrace) return
         val braceSpace = node.findChildBefore(beforeType, WHITE_SPACE)
-        if (!(braceSpace == null || braceSpace.elementType != WHITE_SPACE)) {
+        if (braceSpace != null) {
             if (braceSpace.text.contains("\n".toRegex())) {
-                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace", 0){
                 }
+                return
             }
         }
         val nodeText = when (node.elementType) {
             IF -> {
                 when (beforeType) {
-                    THEN -> node.findChildByType(THEN)?.text
-                    else -> node.findChildByType(ELSE)?.text
+                    THEN -> node.findChildByType(THEN)?.findChildByType(BLOCK)?.findChildAfter(LBRACE, WHITE_SPACE)
+                    else -> node.findChildByType(ELSE)?.findChildByType(BLOCK)?.findChildAfter(LBRACE, WHITE_SPACE)
                 }
             }
-            WHEN -> node.findChildBefore(WHEN_ENTRY, WHITE_SPACE)?.text + node.getFirstChildWithType(WHEN_ENTRY)?.text
-            FOR -> node.findChildByType(BODY)?.text
-            else -> node.findChildByType(BLOCK)?.text
+            WHEN -> node.findChildAfter(LBRACE, WHITE_SPACE)
+            FOR, WHILE, DO_WHILE -> node.findChildByType(BODY)?.findChildByType(BLOCK)?.findChildAfter(LBRACE, WHITE_SPACE)
+            CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildAfter(LBRACE, WHITE_SPACE)
+            else -> node.findChildByType(BLOCK)?.findChildAfter(LBRACE, WHITE_SPACE)
         }
-        if (nodeText != null && nodeText[1] != '\n') {
-            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
+        if (nodeText != null && !nodeText.text.contains("\n".toRegex())) {
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect line after opening brace", 0) {
             }
         }
     }
 
-    private fun checkCloseBrace(node: ASTNode) {
-        val space = node.getChildren(null).findLast { it.elementType == WHITE_SPACE }
-        if (!space!!.text.contains("\n".toRegex())) {
-            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "close braces", 0) {
+    private fun checkCloseBrace(node: ASTNode?, configuration: BlockStructureBracesConfiguration) {
```

Comment:
Are there any real cases when you pass `null` as `node` here? Could you make this parameter non-nullable and check nullability outside of this function?